### ISSUE TITLE
webpack 2 crashes on "modules": false with no information

### ISSUE THREAD

**Issue by goldylucks** [webpack/webpack#4012]
**What is the current behavior?**
```bash
ERROR in   TypeError: __webpack_require__(...) is not a function
  
  
  - index.ejs:1205 Object.../node_modules/webpack/buildin/harmony-module.js
    /home/adam/apps/memory-n-back/web/index.ejs:1205:1
  
  - index.ejs:622 __webpack_require__
    /home/adam/apps/memory-n-back/web/index.ejs:622:30
  
  - index.ejs:46 fn
    /home/adam/apps/memory-n-back/web/index.ejs:46:20
  
  
  - index.ejs:1197 Object.../node_modules/webpack/buildin/global.js
    /home/adam/apps/memory-n-back/web/index.ejs:1197:1
  
  - index.ejs:622 __webpack_require__
    /home/adam/apps/memory-n-back/web/index.ejs:622:30
  
  - index.ejs:46 fn
    /home/adam/apps/memory-n-back/web/index.ejs:46:20
  
  
  - index.ejs:1189 Object.../node_modules/lodash/lodash.js
    /home/adam/apps/memory-n-back/web/index.ejs:1189:1
```

**If the current behavior is a bug, please provide the steps to reproduce.**
checking out [this commit](https://github.com/goldylucks/memory-n-back/commit/6f0e93247c62b9c005f181abfaf1957eeb017696) from my [repo](https://github.com/goldylucks/memory-n-back)  and running `npm start`

The thing is I don't know what is causing the crash, because as u can c above, the logs are not very informational, hence I'm opening this issue and not asking on SO

**What is the expected behavior?**
Work with tree shaking and not crashing

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
OS: `ubuntu 16.04`
webpack:` 2.1.0-beta.25`
webpack-dev-server: `2.1.0-beta.0`

it crashes on both webpack and dev server

**Comment by rosskevin** [comment]
@goldylucks - I've run into the same and I've been meticulously trying experiments one at a time.  I finally got a good compilation by changing `transform-runtime` [based on this SO suggestion](http://stackoverflow.com/a/39502362/2363935).  I notice you have it in your babel file as well.

Here is my resulting `.babelrc` that is currently working:

```
{
  "presets": [
    "es2017",
    "es2016",
    ["es2015", {"modules": false}],
    "react",
    "stage-0"
  ],
  "plugins": [
    "babel-relay-plugin-loader",
    "flow-react-proptypes",
    "transform-flow-strip-types",
    "lodash",
    ["transform-runtime",
      {
        "helpers": false,
        "polyfill": false,
        "regenerator": true
      }
    ],
    "react-hot-loader/babel"
  ]
}
```

I don't have any information as to why this is affecting this error, but I wanted to pass on this information in case it can be helpful.

**Comment by rosskevin** [comment]
More information as I experiment with `transform-runtime` plugin settings.

Turning on `helpers`:
```    
["transform-runtime",
  {
    "helpers": true,
    "polyfill": false,
    "regenerator": true
  }
],
```    

yields

```    
Uncaught ReferenceError: exports is not defined
    at Object../node_modules/babel-runtime/helpers/typeof.js (typeof.js:3)
    at __webpack_require__ (bootstrap 0be4201…:686)
    at fn (bootstrap 0be4201…:105)
    at Object../node_modules/url/url.js (url.js:1)
    at __webpack_require__ (bootstrap 0be4201…:686)
    at fn (bootstrap 0be4201…:105)
    at Object.<anonymous> (client:2)
    at Object../node_modules/webpack-dev-server/client/index.js?http:/localhost:3808 (application.js:176590)
    at __webpack_require__ (bootstrap 0be4201…:686)
    at fn (bootstrap 0be4201…:105)
```    
    
Turning on `polyfill`:    
    
```    
["transform-runtime",
  {
    "helpers": false,
    "polyfill": true,
    "regenerator": true
  }
],
```

yields:
    
```    
Uncaught TypeError: $export is not a function
    at Object../node_modules/babel-runtime/node_modules/core-js/library/modules/es6.object.define-property.js (es6.object.define-property.js:3)
    at __webpack_require__ (bootstrap c5fe614…:686)
    at fn (bootstrap c5fe614…:105)
    at Object../node_modules/babel-runtime/node_modules/core-js/library/fn/object/define-property.js (define-property.js:1)
    at __webpack_require__ (bootstrap c5fe614…:686)
    at fn (bootstrap c5fe614…:105)
    at Object../node_modules/babel-runtime/core-js/object/define-property.js (define-property.js:1)
    at __webpack_require__ (bootstrap c5fe614…:686)
    at fn (bootstrap c5fe614…:105)
    at Object../node_modules/babel-runtime/node_modules/core-js/library/modules/_object-dp.js (application.js:42490)        
```

**Comment by goldylucks** [comment]
Works!
Thanks @rosskevin !

PS. it's TL;DR, and it best added at the beginning ;)

**Comment by goldylucks** [comment]
I wonder if webpack 2 guys should document it somewhere?

**Comment by vinnymac** [comment]
@goldylucks I am not sure what tool you use to install your dependencies. From my research it does appear to be package manager specific. Searching for duplicate `babel-runtime` dependencies in your tree and removing any dupes alleviates this.

Yarn has had a bug for a while where it will install `devDependencies` when it isn't told to, such as `yarn install --production`. Even if you don't use yarn if something installs `babel-runtime` when you do not intend for it to this will happen.

Your fix above does indeed work, but I don't think this change should be necessary and actually made my package size increase when I made this change.

For me all I had to do was find the duplicate `babel-runtime` dependency. Remove it, and then rebuild. Yarn [claims](https://github.com/yarnpkg/yarn/issues/2819) to have fixed it, but it does not yet work for me. I thought I would at least inform others about this in case they are seeing the same issue with Webpack.

TL;DR I don't think this is a Webpack issue but more a babel + package manager issue.

**Comment by goldylucks** [comment]
@vinnymac thanks for the update
I was using NPM, now switched to yarn, closing this for now.
